File: Java/src/1678_GoalParserInterpretation/GoalParserInterpretation.java
class Solution {
    public String interpret(String command) {
        String ans = "";
        for(int i = 0 ; i < command.length(); i++){
            if(command.charAt(i) == 'G'){
                ans+="G";
            }else if(command.charAt(i) == '('){
                if(command.charAt(i+1) == ')'){
                    ans+="o";
                }else {
                    ans+="al";
                }
            }
        } 
        return ans ;
    }
}

File: golang/1678_GoalParserInterpretation/goalParserInterpretation.go
package goalparserinterpretation

func interpret(command string) string {
	ans := []byte{}
	for i := 0; i < len(command); i++ {
		if command[i] == 'G' {
			ans = append(ans, 'G')
		} else if command[i] == '(' {
			if command[i+1] == ')' {
				ans = append(ans, 'o')
			} else {
				ans = append(ans, 'a')
				ans = append(ans, 'l')
			}
		}
	}
	return string(ans)
}
